lib/rules/jsx-no-useless-fragment.js

```diff
@@ -19,7 +19,7 @@ function isJSXText(node) {
  * @returns {boolean}
  */
 function isOnlyWhitespace(text) {
-  return text.trim().length === 0;
+  return !/\S/.test(text);
```

Comment:
these aren't necessarily always the same thing; `trim` can be polyfilled to be more correct, but regexes can't be.

Additionally, is this better than `/^[^\S]*$/.test(text)`, which might be clearer?

Reply:
I guess I'm not convinced that `.trim()` is meaningfully slower than using a custom regex. Do you have data to support that?

Reply:
Sorry about the late reply!

I was really convinced a regex would be faster because there'd be less work (stop the moment non-whitespace is found, rather than doing the work of constructing a new string, then checking its length), but after constructing some basic micro benchmarks to check it, they're basically equivalent in speed, and in Chrome 91 + latest V8 sometimes trim was faster.

Thanks for pushing back on this, I'm glad it was double checked - will revert this change.